### PR TITLE
Do not strip trailing slashes when using Reactive REST Client

### LINKED ISSUE #27050: Remote servers that require a trailing slash can no longer be called with Quarkus 2.11
### Describe the bug

As [discussed on Zulip](https://quarkusio.zulipchat.com/#narrow/stream/187030-users/topic/2.2E11.3A.20trailing.20slash.20in.20rest-client/near/291546201), Quarkus 2.11 introduces a change in `quarkus-rest-client-reactive`: trailing slashes are removed from REST paths. This makes it impossible to call a remote server that requires a trailing slash.

### Expected behavior

Trailing slashes should be preserved.

### Actual behavior

Trailing slashes are removed.

### How to Reproduce?

_No response_

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

2.11.0.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by geoand**
Fixes: #27050

### CODE REVIEW

File: independent-projects/resteasy-reactive/server/processor/src/main/java/org/jboss/resteasy/reactive/server/processor/ServerEndpointIndexer.java [hunk 1]
```diff
@@ -381,8 +381,9 @@ protected void handleArrayParam(Map<String, String> existingConverters, String e
         builder.setConverter(new ArrayConverter.ArraySupplier(converter, elementType));
     }
 
-    protected void handlePathSegmentParam(ServerIndexedParameter builder) {
```
Comment (gastaldi) [inline independent-projects/resteasy-reactive/server/processor/src/main/java/org/jboss/resteasy/reactive/server/processor/ServerEndpointIndexer.java:384]: Is this method supposed to be removed?

Comment (geoand) [reply independent-projects/resteasy-reactive/server/processor/src/main/java/org/jboss/resteasy/reactive/server/processor/ServerEndpointIndexer.java:384]: Ouch... No...

Great catch!

Comment (geoand) [reply independent-projects/resteasy-reactive/server/processor/src/main/java/org/jboss/resteasy/reactive/server/processor/ServerEndpointIndexer.java]: Fixed, thanks for spotting this dump mistake on my part :)

Comment (gastaldi) [reply independent-projects/resteasy-reactive/server/processor/src/main/java/org/jboss/resteasy/reactive/server/processor/ServerEndpointIndexer.java]: No problem, that's what I am here for 😆

**Review by gastaldi — APPROVED** [review]

**Pull request merged**